portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,71 @@ function addRandomArloMessage() {
   const arloTalksContainer = document.getElementById('arlo-talks-container');
   arloTalksContainer.innerText = arloMessage;
 }
+
+/**
+ * Shrinks the navbar when user scrolls down.
+ */
+window.onscroll = function() {shrinkNavBar()};
+
+function shrinkNavBar() {
+  if (document.body.scrollTop > 40 || document.documentElement.scrollTop > 40) {
+    document.getElementById("navbar").style.paddingTop = "15px";
+    document.getElementById("navbar").style.paddingBottom = "15px";
+    document.getElementById("navbar-title").style.fontSize = "18px";
+    document.getElementById("navbar-right").style.top = "0px";
+  } else {
+    document.getElementById("navbar").style.paddingTop = "35px";
+    document.getElementById("navbar").style.paddingBottom = "35px";
+    document.getElementById("navbar-title").style.fontSize = "30px";
+    document.getElementById("navbar-right").style.top = "5px";
+  }
+}
+
+/**
+ * Smooths scrolls to About Me Section / top of document
+ */
+function scrollToAboutMe() {
```

Comment:
Can you combine these four functions into a single one that takes the element to scroll to, and perhaps an additional offset, as arguments, then pass those arguments in the HTML `onclick` attributes? That would remove the repetition in this file and make it easier to maintain. I see that `scrollToAboutMe` uses `window.pageYOffset`, but I suspect it's more-or-less a constant offset away from `document.getElementById("aboutme").getBoundingClientRect().top`, which makes it look like the others.

Reply:
I really appreciate all these suggestions, this does makes things a lot cleaner.